waku/wdd.md

```diff
@@ -0,0 +1,75 @@
+# Waku DNS Discovery
+
+> Version: 0.1.0 (Draft)
+>
+> Authors: Dean Eigenmann <dean@status.im>
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Specification](#specification)
+    1. [Records](#records)
+    2. [Traversing](#traversing)
+3. [Footnotes](#footnotes)
+4. [Copyright](#copyright)
+
+## Abstract
+
+In this specification, we describe the DNS discovery protocol used by [Waku](./waku.md). 
+
+## Specification
+
+This specification is based on the DNS discovery proposal for Ethereum<sup>1</sup>. It traverses a merkle tree that is stored in the DNS records of a chosen domain.
+
+### Records
+
+The node lists are encoded as merkle trees and stored within DNS TXT records. The root of the tree is a `TXT` record:
+
+```
+tree-root:v1 r=<root> l=<link-root> seq=<sequence-number> sig=<signature>
+```
+
+- `root`: the root hash of the subtree.
+- `link-root`:  the root hash of subtrees containing links for different subtrees.
+- `sequence-number`: the update sequence number, as an integer.
+- `signature`: a 65-byte secp256k1 EV signature of the `sha256` hash of `root`, `link-root`, and `sequence-number` concatenated.
```

Comment:
If we want to change signature, what would that look like? I suppose the v1 is version number (not specified)?

Reply:
From Disc v5

> Node identity crypto is extensible, use of secp256k1 keys isn't strictly required.

Reply:
@oskarth If we really are going the libp2p route what we could also do is use multihashes in most areas. There sadly is not multisignature format yet though cause it was never accepted.

Reply:
Disc v5 manages to the secp256k key not strictly required, how is that done?